src/coreclr/jit/gentree.cpp

```diff
@@ -21423,45 +21423,84 @@ GenTree* Compiler::gtNewSimdBinOpNode(
             }
             else if (varTypeIsLong(simdBaseType))
             {
-                assert((simdSize == 16) || (simdSize == 32) || (simdSize == 64));
+                if ((simdSize == 32) || compOpportunisticallyDependsOn(InstructionSet_SSE41))
```

Comment:
It looks like we're missing handling around `simdSize == 64`?

Reply:
`simdSize == 64` implies `vpmullq` is available, so we won't end up in this emulation path.  Notice the old assert includes 64 and then immediately below it asserts that simdSize is 16 or 32 :smile:

There are a lot of those confusing asserts around, so I've just been deleting them.  I can comment those instead if you prefer.

Reply:
We generally want to explicitly keep asserts where possible, because they help catch issues when users start refactoring or adding new paths.

That is, being explicit that `assert(simdSize != 64);` helps more than a user hitting the assert a bit further down. Having an attached comment explaining that its unnecessary to handle because `GetHWIntrinsicIdForBinOp` should have returned the singular instruction for `TYP_LONG` helps with that, especially given that we do sometimes hit the overall fallback for GT_MUL on SIMD64 nodes for other base types.